libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/trip/session/MapboxTripSession.kt

```diff
@@ -149,43 +148,54 @@ class MapboxTripSession(
     override fun getState(): TripSessionState = state
 
     /**
-     * Start MapboxTripSession
+     * Starts map matching and starts a foreground service. @see [startFreeDrive]
+     * to follow current location without map matching.
      */
-    override fun start() {
+    override fun startTripSession() {
         if (state == TripSessionState.STARTED) {
             return
         }
         tripService.startService()
-        startLocationUpdates()
+        rawLocationProvider.requestTripSessionUpdates { rawLocation ->
+            mapMatchLocation(rawLocation)
+        }
         state = TripSessionState.STARTED
     }
 
-    private fun startLocationUpdates() {
-        locationEngine.requestLocationUpdates(
-            locationEngineRequest,
-            locationEngineCallback,
-            Looper.getMainLooper()
-        )
-        locationEngine.getLastLocation(locationEngineCallback)
-    }
-
     /**
-     * Stop MapboxTripSession
+     * Stop map matching and the foreground service. Note that [startFreeDrive] will
+     * be automatically re-enabled if it was called.
      */
-    override fun stop() {
+    override fun stopTripSession() {
         if (state == TripSessionState.STOPPED) {
             return
         }
         tripService.stopService()
-        stopLocationUpdates()
+        rawLocationProvider.stopTripSessionUpdates()
         ioJobController.job.cancelChildren()
         mainJobController.job.cancelChildren()
         reset()
         state = TripSessionState.STOPPED
     }
 
-    private fun stopLocationUpdates() {
-        locationEngine.removeLocationUpdates(locationEngineCallback)
+    /**
+     * Start location updates without a service. Consider that [stopTripSession]
+     * takes priority - meaning this will disable when the session is active.
+     */
+    override fun startLocationUpdates() {
+        rawLocationProvider.requestLocationUpdates { rawLocation ->
+            locationObservers.forEach { it.onRawLocationChanged(rawLocation) }
+            CoroutineScope(Dispatchers.IO).launch {
+                val currentDate = Date()
+                navigator.updateLocation(rawLocation, currentDate)
+                val status = navigator.getStatus(currentDate)
+                updateEnhancedLocation(status.enhancedLocation, status.keyPoints)
```

Comment:
interesting behavior difference between these location updates and the trip session. these location updates do not have the status polling loop, they only get status from the raw location.

it may be required to have a polling status loop to achieve good results.

Reply:
left the status polling loop out because this has been a discussion point. we've considered removing it, having a way to receive location updates without it, would give the discussion results to look at.